### DISCUSSION TITLE
Implement an observer using FO's extensibility model

### DISCUSSION THREAD

**Discussion by deiv061** [microsoft/service-fabric-observer#284] [Q&A]
Hello
I'm using the observer with more or less default settings, I've launched AppObserver AppObserverEnableTelemetry, also we're watching for disk Usage CPU, Memory, etc.
It seems to work because it sends data like warning errors and shows up in Service Fabric Explorer. I've been trying since yesterday I was curious about how to run Custom Observer and I looked at your example and the SampleObserverPlugin project. You gave a great example and the project that was made, I wanted to test it before I started doing something to get to know a little more. I copied the classes and after the build, I put the DLL in the Plugin folder as well(trying with/without jsons and pdbs.). But I don't see anything to be sent to in the App insight, I also tried Log Analytics in Settings.xml. I also ran(commentOut) the configuration for the plugin in Settings.xml. It also doesn't show in the ServiceFbaric ETW event something to indicate that the plugin is working as expected. I'm a little stuck, I'm missing something I guess. 
 I would also appreciate it if you could share some experience on how I can debug the whole thing so I can play around a bit to see how the plugin works. I want to make a plugin that will Observe about 40 AppTypes each one has 7 Stateless services and 1 Stateful, I want to monitor and collect some performance counters like their CPU and Memory and after each iteration of the FabricObserver send that perfCounter to Application Insights. I would analyze for spikes and peaks of our services themselves and if they have any memory leaks. Thank you! 
 
 **ServiceFabric:10.1.1541.9590
 FabricObserver:3.2.11**
![Screenshot_195](https://github.com/microsoft/service-fabric-observer/assets/25546812/69cc2576-d364-4cb5-bd58-b318d73ad190)

**Comment by deiv061** [comment]
@GitTorre
Can you assist :)

**Comment by GitTorre** [comment]
The issue is that you are using 8.0 version of DependencyInjection.Abstractions, but FO ships with version 7.0. That means, an IOException will be thrown because the loader can't locate the 8.0 version because it is not part of the FO code package. This problem doesn't exist if you use the Extensibility library to build your plugins as FO dependencies like this are already in place in that package. By the way, IOException is ignored in FO's plugin loader impl in FabricObserver.cs. That should be changed to make this type of issue more obvious. So, we will add a work item to not hide that issue from customers. Thanks for reporting this :)
